src/ScreenshotManager.vala

```diff
@@ -162,8 +161,14 @@ namespace Gala {
 #else
             Cairo.RectangleInt clip = { rect.x - (int) actor_x, rect.y - (int) actor_y, rect.width, rect.height };
 #endif
-            var image = (Cairo.ImageSurface) window_texture.get_image (clip);
+            var image = (Cairo.ImageSurface) window_actor.get_image (clip);
             if (include_cursor) {
+                if (window.get_client_type () == Meta.WindowClientType.WAYLAND) {
+                    float resource_scale = window_actor.get_resource_scale ();
+
+                    image.set_device_scale (resource_scale, resource_scale);
```

Comment:
because there is a difference between the cursor and window scale, it doesn't really matter if it is only the window that is in the screenshot (and that's how GNOME Shell is doing it :see_no_evil:)